src/vivarium_gates_nutrition_optimization_child/components/observers.py

```diff
@@ -92,111 +111,102 @@ def on_post_setup(self, event: Event) -> None:
-#         builder.value.register_value_modifier(
-#             self.metrics_pipeline_name,
-#             modifier=self.metrics,
-#             requires_columns=["age", "exit_time", "alive"],
-#         )
-#
-#     ########################
-#     # Event-driven methods #
-#     ########################
-#
-#     def on_collect_metrics(self, event: Event) -> None:
-#         pop = self.population_view.get(event.index)
-#         pop_born = pop[pop["entrance_time"] == event.time - event.step_size]
-#
-#         if pop_born.empty:
-#             return
-#         groups = self.stratifier.group(
-#             pop_born.index, self.config.include, self.config.exclude
-#         )
-#         for label, group_mask in groups:
-#             pop_born_in_group = pop_born[group_mask]
-#             low_birth_weight_mask = (
-#                 pop_born_in_group[self.birth_weight_column_name] < self.low_birth_weight_limit
-#             )
-#             new_observations = {
-#                 f"live_births_{label}": pop_born_in_group.index.size,
-#                 f"birth_weight_sum_{label}": pop_born_in_group[
-#                     self.birth_weight_column_name
-#                 ].sum(),
-#                 f"gestational_age_sum_{label}": pop_born_in_group[
-#                     self.gestational_age_column_name
-#                 ].sum(),
-#                 f"low_weight_births_{label}": low_birth_weight_mask.sum(),
-#             }
-#             self.counts.update(new_observations)
-#
-#     ##################################
-#     # Pipeline sources and modifiers #
-#     ##################################
-#
-#     def metrics(self, index: pd.Index, metrics: Dict) -> Dict:
-#         metrics.update(self.counts)
-#
-#         return metrics
+class BirthObserver:
+
+    configuration_defaults = {
+        "stratification": {
+            "birth": {
+                "exclude": [],
+                "include": [],
+            }
+        }
+    }
+
+    def __repr__(self):
+        return "BirthObserver()"
+
+    ##############
+    # Properties #
+    ##############
+
+    @property
+    def name(self):
+        return "birth_observer"
+
+    #################
+    # Setup methods #
+    #################
+
+    # noinspection PyAttributeOutsideInit
+    def setup(self, builder: Builder) -> None:
+        self.clock = builder.time.clock()
+        self.config = builder.configuration.stratification.birth
+        self.birth_weight_column_name = "birth_weight_exposure"
+        self.gestational_age_column_name = "gestational_age_exposure"
+        self.low_birth_weight_limit = 2500  # grams
+
+        columns_required = [
+            "entrance_time",
+            self.birth_weight_column_name,
+            self.gestational_age_column_name,
+        ]
+        self.population_view = builder.population.get_view(columns_required)
+
+        builder.results.register_observation(
+            name=f"live_births",
+            aggregator=self.count_live_births,
```

Comment:
You need a filter for `"alive == 'alive'"` for all of these, since stillborn children will appear in the population (and they should be initialized as dead). 

Initializing stillborn children as dead should be completed in a separate ticket (and I should have called that out explicitly before).